test/e2e/simple.go

```diff
@@ -50,10 +50,8 @@ func E2ESimple(logger *log.Logger) error {
 	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
 	defer cancel()
 	opts := txsim.DefaultOptions().WithSeed(seed).SuppressLogs()
-	err = txsim.Run(ctx, testnet.GRPCEndpoints()[0], kr, encCfg, opts, sequences...)
-
-	if !errors.Is(err, context.DeadlineExceeded) {
-		return fmt.Errorf("expected context.DeadlineExceeded, got %w", err)
+	if err = txsim.Run(ctx, testnet.GRPCEndpoints()[0], kr, encCfg, opts, sequences...); !isDeadlineExceeded(err) {
+		return fmt.Errorf("expected error to contain context deadline exceeded but got %w", err)
```

Comment:
context deadline exceeded is only one type of errors that txsim can return, and there are others. Why are we expecting it to be the only one?

Reply:
if that's the happy path, why would we need to execute the rest of the tests? 

also, why wouldn't it be something like:

```

if err = txsim.Run(ctx, testnet.GRPCEndpoints()[0], kr, encCfg, opts, sequences...); !isDeadlineExceeded(err) {
	return fmt.Errorf("expected error to contain context deadline exceeded but got %w", err)
}

=>

err = txsim.Run(ctx, testnet.GRPCEndpoints()[0], kr, encCfg, opts, sequences...)
assert.Falsef(t, isDeadlineExceeded(err), "expected error to contain context deadline exceeded but got %w", err)
```
or similar?

Reply:
oh I kinda see what you mean. I think you meant:

```go
err = txsim.Run(ctx, testnet.GRPCEndpoints()[0], kr, encCfg, opts, sequences...)
assert.True(t, isDeadlineExceeded(err), "expected error to contain context deadline exceeded but got %w", err)
```

but yea that would make it clearer that the test expects a context deadline exceeded error.